app/models/teacher_set.rb

```diff
@@ -825,11 +825,14 @@ def update_included_book_list(teacher_set_record)
   def update_subjects_via_api(subject_name_array)
     LogWrapper.log('DEBUG', {'message' => 'update_subjects_via_api.start','method' => 'teacher_set.update_subjects_via_api'})
 
-    # teacher_set.rb facets_for_query uses cached results of each query
-    Rails.cache.clear unless Rails.env.test?
-
     return if subject_name_array.blank?
 
+    unless MlnConfigurationController.new.feature_flag_config('teacherset.data.from.elasticsearch.enabled')
```

Comment:
Please put a little more comment, explaining this was necessary in the days before we had ElasticSearch. When do you think we can remove this?